build/common.go

```diff
@@ -223,7 +223,7 @@ func (Build) Backend() error {
 // BuildAll builds production executables for all supported platforms.
 func BuildAll() { //revive:disable-line
 	b := Build{}
-	mg.Deps(b.Linux, b.Windows, b.Darwin, b.DarwinARM64, b.LinuxARM64, b.LinuxARM, b.GenerateManifestFile)
+	mg.SerialDeps(b.Linux, b.Windows, b.Darwin, b.DarwinARM64, b.LinuxARM64, b.LinuxARM, b.GenerateManifestFile)
```

Comment:
I would keep the parallel execution to leverage all cores when building plugins, however your point about duplicate package requests is a fair one.

To address it, I would change it so fetching the packages happens as a separate step before building the executable, which is then done in parallel, like so:

```go
// GoModDownload downloads the packages required by the executable by running "go mod download".
func (Build) GoModDownload() error {
	return sh.Run("go", "mod", "download")
}

// BuildAll builds production executables for all supported platforms.
func BuildAll() { //revive:disable-line
	b := Build{}
	mg.Deps(b.GoModDownload)
	mg.Deps(b.Linux, b.Windows, b.Darwin, b.DarwinARM64, b.LinuxARM64, b.LinuxARM, b.GenerateManifestFile)
}

```

Reply:
Hi again!

Each binary target does `go mod download` under hood (if necessary) so separate download target isn't needed.

Furthermore, each binary target utilizes as much cores as possible (or at most `${GOMAXPROCS}` if specified). Running all of them at once will throttle overall system/container.